src/frontend/http.rs

```diff
@@ -1548,12 +1599,24 @@ SELECT
         // CREATE TABLE queries with arrays, so we don't officially support arrays.
 
         let resp = request()
-            .method("POST")
-            .path("/q")
+            .method(method)
+            .path(path)
             .json(&HashMap::from([("query", query)]))
             .reply(&handler)
             .await;
         assert_eq!(resp.status(), StatusCode::OK);
+        assert_eq!(
+            resp.headers()
+                .get(header::CONTENT_TYPE)
+                .unwrap()
+                .to_str()
+                .unwrap(),
+            "application/json; smallint_val=Int16; integer_val=Int32; bigint_val=Int64; char_val=Utf8; \
+            varchar_val=Utf8; text_val=Utf8; float_val=Float32; real_val=Float32; double_val=Float64; \
+            bool_val=Boolean; date_val=Date32; timestamp_val=Timestamp(Nanosecond, None); \
+            int_array_val=List(Field { name: \"item\", data_type: Int64, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }); \
+            text_array_val=List(Field { name: \"item\", data_type: Utf8, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} })",
```

Comment:
Is there a name for this format of `Type(Params)`? Just wondering how a client might parse it. It looks almost like JSON but not quite.

Reply:
I don't think so, it's really only a serialization of a recursive type enum variant `DataType::List(Box<Field>)`. 

This however sort of breaks the abstraction, since we serialize only the name of the `Field` on the left, but on the right the entire (nested) `Field` struct gets unwrapped.

Reply:
Yeah, I guess the more human-readable verison would be `Int64[]`. Would you lose any information if you forced it to be like that? I guess `nullable` is important.

Related: What would be the type of a scalar column that is nullable? Is it even possible to produce such a query?

Reply:
I was able to construct such a query on my Seafowl test DB:

```
curl -i \
    -H "Content-Type: application/json" \
https://splitgraph-testing-nomad.seafowl.cloud/q -d@-<<EOF
{"query": "SELECT name, value, CASE when value > 2 THEN null WHEN value < 2 THEN 'ok' END islessthantwo FROM test_cache;"}
EOF
```

which returns the rows with nullable results as undefined (this is expected):

```
HTTP/2 200 
content-type: application/octet-stream
vary: Content-Type, Origin, X-Seafowl-Query
content-length: 386
date: Thu, 20 Apr 2023 14:54:11 GMT
server: Fly/ad79467d (2023-04-14)
via: 2 fly.io
fly-request-id: 01GYFJH6RQ0132ZX4000EK6RSY-lhr

{"name":"Alpha","value":2.3}
{"name":"Alpha","value":2.4}
{"name":"Beta","value":9.3}
{"name":"Beta","value":4.0}
{"name":"Gamma","value":6.8}
{"islessthantwo":"ok","name":"Gamma","value":1.2}
{"name":"Alpha","value":2.3}
{"name":"Alpha","value":2.4}
{"name":"Beta","value":9.3}
{"name":"Beta","value":4.0}
{"name":"Gamma","value":6.8}
{"islessthantwo":"ok","name":"Gamma","value":1.2}

```

So what would we expect the column type of `islessthantwo` to be in the header?

Reply:
> What would be the type of a scalar column that is nullable? Is it even possible to produce such a query?

Nullability is being [tracked at the `Field`](https://github.com/apache/arrow-rs/blob/master/arrow-schema/src/field.rs#L40) level, not the data type level, so in the case of a scalar column that is nullable there would be no info about it (as we'd record only the field name).

> There was some code that serializes column types into the database in the arrow-native JSON format if that didn't go away with the delta-rs migration.

That's still there, though it may be too [verbose](https://github.com/splitgraph/seafowl/blob/main/src/schema.rs#L72-L85).

> We could also serialize it into SQL datatypes (TEXT instead of Utf8, BIGINT[] instead of List(...Int64...))

I think this is the best approach overall, as it also keeps the output implementation agnostic, i.e. SQL in SQL out. I'll try to find a way to make this work.

Reply:
We could also (which might be overkill) emit a JSONSchema instead of rolling our own custom format, which supports dates and floating point numbers. Could be useful for clients autogenerating parsers downstream.

Reply:
It would definitely be nice to include nullability in this response header, even if it's technically part of `field.nullable` rather than `field.type`.